### ISSUE TITLE
LDtk crashes when I try to edit the atlas padding of a resized tileset image

### ISSUE THREAD

**Issue by Sheepolution** [deepnight/ldtk#469]
Stack:
```
LDtk version: 0.9.2-64bits
Pixels.sub() outside bounds
Error

```
Processes:
```
#0 App
 |--- #3 page.Editor
       |--- #4 misc.FileWatcher
       |--- #5 display.WorldRender
       |--- #6 display.LevelRender
       |--- #7 display.Camera
       |--- #8 display.Rulers[--]
       |--- #9 tool.SelectionTool[--]
       |--- #10 tool.lt.DoNothing[--]
       |--- #11 WorldTool
       |--- #12 tool.PanView[--]
       |--- #98 tool.lt.TileTool[--]
       |--- #131 ui.modal.panel.EditTilesetDefs
       |--- #144 ui.modal.dialog.TextEditor
       |--- #145 ui.modal.Progress
             |--- #146 dn.Process
```

Log:
```
[GENERAL]      Init pixel data cache for ../images/tileset.png
[GENERAL]      "Initializing pixel data cache for "tileset.png"", 3 operation(s):
[GENERAL]      Done "Initializing pixel data cache for "tileset.png"" (0.02s)
[EVENT]        TilesetDefPixelDataCacheRebuilt 5
[RENDER]       Reset world render
[RENDER]       Rendered world level data.Level.TestLevel(#0)
[FILE]         Changed on disk: [cut]/tileset.png
[CACHE]        Reloading cached image ../images/tileset.png...
[CACHE]        Disposing cached image ../images/tileset.png...
[CACHE]        Caching image ../images/tileset.png...
[FILE]         Reloading tileset: TestTileset path=../images/tileset.png
[FILE]         Loading atlas image: ../images/tileset.png...
[FILE]          -> Loaded 755 bytes.
[FILE]          -> Decoded 54x54 pixels.
[GENERAL]      Tileset remapping...
[GENERAL]       > success
[FILE]          -> Reload result: RemapSuccessful
[FILE]          -> pixelData: need rebuild
[GENERAL]      Init pixel data cache for ../images/tileset.png
[GENERAL]      "Initializing pixel data cache for "tileset.png"", 4 operation(s):
[EVENT]        TilesetDefChanged 5
[GENERAL]      Init pixel data cache for ../images/tileset.png
[GENERAL]      "Initializing pixel data cache for "tileset.png"", 4 operation(s):
[GENERAL]      Done "Initializing pixel data cache for "tileset.png"" (0.24s)
[EVENT]        TilesetDefPixelDataCacheRebuilt 5
[RENDER]       Reset world render
[GENERAL]      Done "Initializing pixel data cache for "tileset.png"" (0.22s)
[EVENT]        TilesetDefPixelDataCacheRebuilt 5
[RENDER]       Reset world render
[RENDER]       Rendered world level data.Level.TestLevel(#0)
[FILE]         Reading HTML template editTilesetDefs from cache
[EVENT]        TilesetDefChanged 5
[GENERAL]      Init pixel data cache for ../images/tileset.png
[GENERAL]      "Initializing pixel data cache for "tileset.png"", 4 operation(s):
[GENERAL]      Done "Initializing pixel data cache for "tileset.png"" (0.03s)
[EVENT]        TilesetDefPixelDataCacheRebuilt 5
[RENDER]       Reset world render
[RENDER]       Rendered world level data.Level.TestLevel(#0)
[EVENT]        ToolOptionChanged
[EVENT]        TilesetDefChanged 5
[GENERAL]      Init pixel data cache for ../images/tileset.png
[GENERAL]      "Initializing pixel data cache for "tileset.png"", 2 operation(s):
[GENERAL]      Page started: page.CrashReport()
[FILE]         Loading page template: crashReport from [cut]/AppData/Local/Programs/ldtk/resources/app.asar/assets/tpl/pages/crashReport.html
[ERROR]        Pixels.sub() outside bounds (Error)
[ERROR]        undefined

[GENERAL]      
#0 App
 |--- #3 page.Editor
       |--- #4 misc.FileWatcher
       |--- #5 display.WorldRender
       |--- #6 display.LevelRender
       |--- #7 display.Camera
       |--- #8 display.Rulers[--]
       |--- #9 tool.SelectionTool[--]
       |--- #10 tool.lt.DoNothing[--]
       |--- #11 WorldTool
       |--- #12 tool.PanView[--]
       |--- #98 tool.lt.TileTool[--]
       |--- #131 ui.modal.panel.EditTilesetDefs
       |--- #144 ui.modal.dialog.TextEditor
       |--- #145 ui.modal.Progress
             |--- #146 dn.Process
 |--- #147 page.CrashReport

[FILE]         Cleared all file watches
```

**Comment by deepnight** [comment]
Hi!

Could you please describe how exactly this crash happened?

**Comment by Sheepolution** [comment]
I did a bit more research on this. This is what I found.

For this I used the following image:
![tileset](https://user-images.githubusercontent.com/2232780/115905705-c42f8480-a466-11eb-93a8-21b51ce56f96.png)

Steps to reproduce
1. Create a new tileset using the image above.
2. Make the layout 16 px tiles, spaced by 2 px, with an atlas padding of 1 px. I'm not sure how important these values are for the bug to happen.
3. Use an image editor to resize the image. Increase its size by a few pixels (too small for another 16 tile to fit)
4. After the image has automatically updated new invisible tiles have bene been added. Click on one of those tiles to let LDtk crash.


![image](https://user-images.githubusercontent.com/2232780/115906330-aca4cb80-a467-11eb-9421-e62694261604.png)
